src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1424,6 +1424,243 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
+                                  Permissions: permissions,
+                                  Relationships: null,
+                                  Mappings: null);
+
+            string entityName = "Stock";
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions, entity, entityName);
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(CUSTOM_CONFIG, configuration.ToJson());
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                // A create mutation operation is executed in the context of Anonymous role. The Anonymous role has create action configured but lacks
+                // read action. As a result, a new record should be created in the database but the mutation operation should return an error message.
+                string graphQLMutation = @"
+                    mutation {
+                      createStock(
+                        item: {
+                          categoryid: 5001
+                          pieceid: 5001
+                          categoryName: ""SciFi""
+                          piecesAvailable: 100
+                          piecesRequired: 50
+                        }
+                      ) {
+                        categoryid
+                        pieceid
+                      }
+                    }";
+
+                JsonElement mutationResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                    client,
+                    server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                    query: graphQLMutation,
+                    queryName: "createStock",
+                    variables: null,
+                    clientRoleHeader: null
+                    );
+                Assert.IsTrue(mutationResponse.ToString().Contains("The mutation operation was successful but the current user is unauthorized to view the response due to lack of read permissions"));
+
+                // pk_query is executed in the context of Authenticated role to validate that the create mutation executed in the context of Anonymous role
+                // resulted in the creation of a new record in the database.
+                string graphQLQuery = @"
+                    {
+                      stock_by_pk(categoryid: 5001, pieceid: 5001) {
+                        categoryid
+                        pieceid
+                        categoryName
+                      }
+                    }";
+
+                string authToken = AuthTestHelper.CreateStaticWebAppsEasyAuthToken();
+
+                JsonElement queryResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
```

Comment:
this next section is now a new test, I would break this into a separate test method. New role execution context, running a GraphQL query to validate db data which results in different assertion expectations.

Reply:
I've introduced a helper method to perform this validaton